core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -376,28 +376,19 @@ trait WorkspaceComponent {
       filter(rec => (rec.namespace === namespaceName.value))
     }
 
-    def listOutdatedEntityCaches(limit: Int): ReadAction[Seq[(UUID, Timestamp)]] = {
-      //We don't necessarily want all of the workspaces back, just the ones that we can handle in one pass
-      //We also order by lastModified so we can handle the ones that are the longest out of date
-      //0 is a "magic" timestamp that means that the cache should not be updated because it produces
-      //an error that breaks the statistics cache monitor. These are unexpected cases that would
-      //need manual intervention.
-      filter(rec => (rec.entityCacheLastUpdated < rec.lastModified && rec.entityCacheLastUpdated > new Timestamp(0))).take(limit).sortBy(_.lastModified.asc).map { ws => (ws.id, ws.lastModified) }.result
+    def findMostOutdatedEntityCacheAfter(timestamp: Timestamp): ReadAction[Option[(UUID, Timestamp)]] = {
+      // Find the workspace that has the entity cache that is the most out of date:
+      // A. Workspace has a cacheLastUpdated date that is not current ("current" means equal to lastModified)
+      // B. cacheLastUpdated is after @param timestamp
+      // C. Ordered by lastModified from oldest to newest. Meaning, return the workspace that was modified the longest ago
+      uniqueResult[(UUID, Timestamp)](filter(rec => rec.entityCacheLastUpdated < rec.lastModified && rec.entityCacheLastUpdated > timestamp).sortBy(_.lastModified.asc).take(1).map { ws => (ws.id, ws.lastModified) })
```

Comment:
I'm inclined to push this off until the metrics follow-up work, and decide then if we should adopt this. I don't think this will have much effect in practice but I do think it could be useful for returning slightly more accurate metrics which is why I want to hold off until then.